nx/lib/nx.ex

```diff
@@ -3586,14 +3586,53 @@ defmodule Nx do
       iex> Nx.compatible?(%{foo: Nx.iota({3, 2})}, %{bar: Nx.iota({3, 2})})
       false
 
+  ## Vectorized tensors
+
+  Same compatibility criteria applies to vectorized tensors, but there's
+  the additional requirement that if a vectorized axis name appears in both
+  tensors, it must appear with the same size, or one of them must be 1.
```

Comment:
Vectorized axes must be the same size. We don't consider shape `{1}` to be compatible with `{10}`.